fix: Update Oracle to new docs style (#1251)
appsmithorg/appsmith-docs: website/docs/reference/datasources/querying-oracle.md

---
sidebar_position: 11
description: Connect Appsmith to an Oracle database and create queries.
---

# Oracle

> This page describes how to connect your application to your Oracle databases and use queries to manage its content.

Reviewer: let's use the updated line

---
sidebar_position: 11
description: Connect Appsmith to an Oracle database and create queries.
---

# Oracle

> This page gives information to connect Appsmith to an Oracle database and to query its data in your applications.

Reviewer: ```suggestion
This page gives information to connect Appsmith to an Oracle database and to and to read and write data in your applications.
```

> ## Connection parameters

Reviewer: If you check PostgreSQL, there is a missing heading above this one and also the callout should be moved here too.

The following is a reference guide that provides a description of the parameters for connecting to an Oracle database.

<figure>
  <img src="/img/oracle-datasource-config.png" style={{width: "100%", height: "auto"}} alt="Configuring an Oracle datasource." />
  <figcaption align="center"><i>Configuring an Oracle datasource.</i></figcaption>
</figure>

:::caution important
If you are a self-hosted user, you must whitelist the IP addresses 18.223.74.85 and 3.131.104.27 of the Appsmith deployment on your database instance. See [Managing Internet Protocol Allowlist and Blocklist Rules](https://docs.oracle.com/en/cloud/get-started/subscriptions-cloud/mmocs/managing-internet-protocol-whitelist-and-blacklist-rules.html) for more details.
:::

<dl>
  <dt><b>Host Address</b></dt>
  <dd>The network location of your Oracle database. This can be a domain name or an IP address. To connect to a local Oracle database, see <a href="/advanced-concepts/more/how-to-work-with-local-apis-on-appsmith"><b>Connect Local Database</b></a> for directions. </dd><br />

  <dt><b>Port</b></dt>

>   <dd>The port number to connect to on the server. </dd><br />

Reviewer: Is there a default port that it connects to? If yes, please mention it

  <dt><b>Service Name</b></dt>
  <dd>The service name for your database instance. </dd><br />

  <dt><b>Authentication</b></dt>

>   <dd>The <b>Username</b> and <b>Password</b> for the user with which you are connecting to the database.</dd><br />

Reviewer: Username and password are different fields and should be described separately.

  <dt><b>SSL Mode</b></dt>
  <dd>Determines whether your queries use an SSL connection to communicate with the database.</dd><br />
  <dd><i>Options:</i>
    <ul>

>       <li><b>TLS:</b> Appsmith attempts to connect using TLS.</li>

Reviewer: these look incomplete. Can we use the one given in the draft. When we say attempt in TLS, it could mean that the encryption is either required or allowed. There shouldn't be any ambiguity here.

>       <li><b>Disabled:</b> Appsmith won't attempt any sort of SSL connection.</li>

Reviewer: ```suggestion
      <li><b>Disabled:</b> Disables SSL completely, and all connections will be established without encryption.</li>
```

    </ul>
  </dd>
</dl>

## Query Oracle

The following section provides examples of creating basic CRUD queries for Oracle.

:::info

> For reference on Oracle's flavor of SQL syntax, see the official [**Oracle SQL documentation**](https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/Basic-Elements-of-Oracle-SQL.html#GUID-41D065C3-3449-4DAE-B2D8-4DF256FFC88A).

Reviewer: ```suggestion
For Oracle SQL syntax, see the official documentation [**SQL Language Reference**](https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/Basic-Elements-of-Oracle-SQL.html#GUID-41D065C3-3449-4DAE-B2D8-4DF256FFC88A).
```

:::

> ### Select

Reviewer: ```suggestion
### Fetch Data
```

Reviewer: Can we please change the others also similar to PostgreSQL

> If you want to connect to a locally hosted database, you can use a service like ngrok to expose it. For more information, see [How to connect to local database on Appsmith](/advanced-concepts/more/how-to-work-with-local-apis-on-appsmith).

Reviewer: ```suggestion

```

Reviewer: as we are mentioning it right below in the host parameter

:::

<dl>
  <dt><b>Host Address</b></dt>
  <dd>The network location of your Oracle database. This can be a domain name or an IP address. To connect to a local Oracle database, see <a href="/advanced-concepts/more/how-to-work-with-local-apis-on-appsmith"><b>Connect Local Database</b></a> for directions. </dd><br />

  <dt><b>Port</b></dt>
  <dd>The port number to connect to on the server. </dd><br />

  <dt><b>Service Name</b></dt>
  <dd>The service name for your database instance. </dd><br />

  <dt><b>Authentication</b></dt>
  <dd>The <b>Username</b> and <b>Password</b> for the user with which you are connecting to the database.</dd><br />

  <dt><b>SSL Mode</b></dt>
  <dd>Determines whether your queries use an SSL connection to communicate with the database.</dd><br />
  <dd><i>Options:</i>
    <ul>
      <li><b>TLS:</b> Appsmith attempts to connect using TLS.</li>
      <li><b>Disabled:</b> Appsmith won't attempt any sort of SSL connection.</li>
    </ul>
  </dd>
</dl>

> :::caution important

Reviewer: this callout has to be right under the Connect Oracle heading.

> If you are a self-hosted user, you may need to whitelist the IP address of the Appsmith deployment on your database instance or VPC before connecting to a database.

Reviewer: ```suggestion
If you are a self-hosted user, you must whitelist the IP addresses 18.223.74.85 and 3.131.104.27 of the Appsmith deployment on your database instance before connecting to a database. See [Managing Internet Protocol Allowlist and Blocklist Rules](https://docs.oracle.com/en/cloud/get-started/subscriptions-cloud/mmocs/managing-internet-protocol-whitelist-and-blacklist-rules.html) for more details.```

Reviewer: @rohan-arthur can you confirm if the link shared above would help users with whitelisting IP addresses? Thanks. cc: @spark-c

> 18.223.74.85 and 3.131.104.27 are the IP addresses of the Appsmith cloud instances that need to be whitelisted.

Reviewer: ```suggestion
```
